### ISSUE TITLE
shadow assoc?

### ISSUE THREAD

**Issue by gilch** [hylang/hy#1396]
@kirbyfan64 attempted this before, #725, but now that `assoc` is a macro #1378 and we expect it will always return `None`, do we want to try to shadow it with an equivalent function? Are there still problems?

**Comment by kirbyfan64** [comment]
Will probably be a lot easier...

**Comment by Kodiologist** [comment]
Having it always return `None` does somewhat reduce the motivation for providing a shadow function.

**Comment by Kodiologist** [comment]
Yeah, it's hard to imagine how it would be useful. Shadow functions are for use in higher-order contexts where you're using return values.
